src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -593,11 +612,56 @@ def docKeyUpdate(self, msOutDoc, **kwargs):
                 self.logger.warning(msg)
         return msOutDoc
 
-    def docInfoUpdate(self, msOutDoc):
+    def docInfoUpdate(self, msOutDoc, pipeLine=None):
         """
         A function intended to fetch and fill into the document all the needed
         additional information like campaignOutputMap etc.
         """
+
+        # Fill the destinationOutputMap first
+        if msOutDoc['isRelVal']:
+            destinationOutputMap = []
+            wflowDstSet = set()
+            updateDict = {}
+            for dataset in msOutDoc['OutputDatasets']:
+                _, dsn, procString, dataTier = dataset.split('/')
+                destination = set()
+                if dataTier != "RECO" and dataTier != "ALCARECO":
```

Comment:
We should check how Unified deals with RECO and ALCARECO datasets.

Reply:
I may need your help here with that. I cannot find anything else than what I have already implemented. Maybe a fresh look on the problem from another person, can reveal things I am constantly overlooking.